### PR TITLE
Migrate controllers to the new Database (closes #15)

### LINKED ISSUE #15: Migrate logic to use new Database
After the DB is ready, replace all mocked methods with real working DB logic

### PR DESCRIPTION
**Pull request by yugisu**

### CODE REVIEW

File: src/main/java/shortener/database/entities/Alias.java [hunk 1]
```diff
@@ -10,4 +10,6 @@ public record Alias(String alias,
                     String url,
                     Long userId,
                     Integer usages) {
+
+  public static final Integer DEFAULT_RANDOM_ALIAS_LENGTH = 5;
```
Comment (IINamelessII) [inline src/main/java/shortener/database/entities/Alias.java:14]: Suggest ALIAS_LENGTH_DEFAULT here (using prefix approach)
+ RANDOM part seems redundant

File: src/main/java/shortener/httphandler/RedirectController.java [hunk 2]
```diff
@@ -31,13 +32,14 @@
...
-    Alias aliasRecord;
     try {
-      aliasRecord = urlsRepository.get(alias);
+      Alias aliasRecord = db.get(db.aliasTable, alias);
+
+      return HttpResponse.redirect(new URI(aliasRecord.url()));
     } catch (NotFound exc) {
       return HttpResponse.notFound("Alias not found.");
+    } catch (IOException exc) {
+      return HttpResponse.serverError();
```
Comment (RedBullDogE) [inline src/main/java/shortener/httphandler/RedirectController.java:42]: What about this parts? It seems like we need some layer for encapsulating this constructions. Should it be Repository or something?

File: src/main/java/shortener/httphandler/UrlController.java [hunk 3]
```diff
@@ -69,34 +72,47 @@
     // Create alias records and check for its uniqueness
     if (alias == null || alias.isBlank()) {
       try {
-        urlsRepository.create(shortenData.url(), userId);
+        String randomAlias = RandomStringGenerator.generate(Alias.DEFAULT_RANDOM_ALIAS_LENGTH);
+
+        db.create(db.aliasTable, new Alias(randomAlias, url, userId, 0));
```
Comment (RedBullDogE) [inline src/main/java/shortener/httphandler/UrlController.java]: For example we can move this logic from controller to <repository>

**Review by IINamelessII — APPROVED** [review]
Nice removal of redundant layer :) Finally, we have 3 layers that were discussed at the start

**Review by RedBullDogE — COMMENTED** [review]
What do you think?

File: src/main/java/shortener/httphandler/UrlController.java (hunk 3 above)
Comment (yugisu) [reply src/main/java/shortener/httphandler/UrlController.java]: @RedBullDogE You are right, will abstract these parts in next iterations. Agreed?

**Pull request merged**